AppKit/CPTextView/CPLayoutManager.j

```diff
@@ -1110,6 +1110,27 @@ var _objectsInRange = function(aList, aRange)
     style.whiteSpace = "pre";
     style.backgroundColor = "transparent";
     style.font = [aFont cssString];
+    
+    if(aUnderline == CPUnderlineStyleSingle)
+    {
+        style.textDecoration = "underline";
+        style.textDecorationStyle = "solid";
+    }
+    if(aUnderline == CPUnderlineStyleDouble)
+    {
+        style.textDecoration = "underline";
+        style.textDecorationStyle = "double";
+    }
+    if(aUnderline == CPUnderlineStylePatternDot)
+    {
+        style.textDecoration = "underline";
+        style.textDecorationStyle = "dotted";
+    }
+    if(aUnderline == CPUnderlineStylePatternDash)
+    {
+        style.textDecoration = "underline";
+        style.textDecorationStyle = "dashed";
+    }
```

Comment:
Why not use a switch statement like this? It will be cleaner, faster and a lot more good looking 😄 

```
    switch (aUnderline)
    {
        case CPUnderlineStyleSingle:
            style.textDecoration = "underline";
            style.textDecorationStyle = "solid";
            break;

        case CPUnderlineStyleDouble:
            style.textDecoration = "underline";
            style.textDecorationStyle = "double";
            break;

        case CPUnderlineStylePatternDot:
            style.textDecoration = "underline";
            style.textDecorationStyle = "dotted";
            break;

        case CPUnderlineStylePatternDash:
            style.textDecoration = "underline";
            style.textDecorationStyle = "dashed";
            break;
    }
```

Reply:
you may factor out the style.textDecoration = "underline"; that is common to all cases